org.eclipse.jdt.debug.tests/tests/org/eclipse/jdt/debug/tests/launching/LongClassPathTests.java

```diff
@@ -242,6 +247,22 @@ protected void setLongClasspath(IJavaProject javaProject, int minClassPathLength
 		javaProject.setRawClasspath(classpathEntries.toArray(new IClasspathEntry[classpathEntries.size()]), null);
 	}
 
+	public static void waitForBuild() {
+		boolean wasInterrupted = false;
+		do {
+			try {
+				Job.getJobManager().join(ResourcesPlugin.FAMILY_AUTO_BUILD, null);
```

Comment:
before join() a job in tests that job should be woke up so that not so long time outs are needed.